packages/dev-server-hmr/src/HmrPlugin.ts

```diff
@@ -228,7 +247,18 @@ export class HmrPlugin implements Plugin {
     }
 
     // Nothing left to try
-    this._broadcast({ type: 'hmr:reload' });
+    if (dependents.size === 0) {
```

Comment:
@43081j I added this to avoid bubbled modules to always trigger a reload, what do you think?